pkg/api/pluginproxy/token_provider_azure.go

```diff
@@ -44,9 +44,15 @@ func getAzureCredentials(cfg *setting.Cfg, authParams *plugins.JwtTokenAuth) azc
 	// * If authType and other fields aren't set then it means the datasource never been configured
 	//   and managed identity is the default authentication choice as long as managed identities are enabled
 	isManagedIdentity := authType == "msi" || (authType == "" && clientId == "" && cfg.Azure.ManagedIdentityEnabled)
+	isUserIdentity := authType == "userid" && cfg.Azure.UserIdentityEnabled
 
 	if isManagedIdentity {
 		return &azcredentials.AzureManagedIdentityCredentials{}
+	} else if isUserIdentity {
+		return &azcredentials.AzureUserIdentityCredentials{
+			TokenEndpoint: cfg.Azure.UserIdentityTokenEndpoint,
+			AuthHeader:    cfg.Azure.UserIdentityAuthHeader,
```

Comment:
This should not be a part of the credential object.

Comment here:
 pkg/tsdb/azuremonitor/azcredentials/credentials.go